runtime/onert/backend/train/BackendContext.cc

```diff
@@ -110,15 +113,26 @@ backend::train::ITensorRegistry *BackendContext::genTrainingTensors()
 {
   const ir::train::TrainableGraph &tgraph = *trainable_graph();
   auto tensor_builder = _tensor_builder;
-
-  tgraph.operands().iterate([&](const ir::OperandIndex &ind, const ir::Operand &obj) {
-    if (external_operands().contains(ind))
+  tgraph.operations().iterate([&](const ir::OperationIndex &, const ir::IOperation &op) {
+    const auto trainable_op = dynamic_cast<const ir::train::TrainableOperation *>(&op);
+    assert(trainable_op);
```

Comment:
The same as in case https://github.com/Samsung/ONE/pull/13048/files#r1620223025. 
```
cannot convert from pointer to base class ‘onert::ir::IOperation’ to pointer to derived class ‘onert::ir::train::TrainableOperation’ because the base is virtual
```